peppy/project.py

```diff
@@ -441,20 +441,67 @@ def templates_folder(self):
         return os.path.join(os.path.dirname(__file__), "submit_templates")
 
 
-    @staticmethod
-    def infer_name(path_config_file):
+    def infer_name(self):
         """
-        Infer project name based on location of configuration file.
+        Infer project name from config file path.
+        
+        First assume the name is the folder in which the config file resides,
+        unless that folder is named "metadata", in which case the project name
+        is the parent of that folder.
+        
+        :param str path_config_file: path to the project's config file.
+        :return str: inferred name for project.
+        """
+        import os
+
+        if hasattr(self, "name"):
+            return(self.name)
+        
+        config_folder = os.path.dirname(self.config_file)
+        project_name = os.path.basename(config_folder)
+        
+        if project_name == "metadata":
+            project_name = os.path.basename(os.path.dirname(config_folder))
+
+        self.name = project_name
+
+        return self.name
+
+    def get_subsample(self, sample_name, subsample_name):
+
+        s = self.get_sample(sample_name)
+        return s.get_subsample(subsample_name)
+
+    def get_sample(self, sample_name):
+        """
+        Get an individual sample object from the project.
+
+        Will raise a ValueError if the sample is not found. In the case of multiple
+        samples with the same name (which is not typically allowed), a warning is
+        raised and the first sample is returned.
+        
+        :param str sample_name: The name of a sample to retrieve
+        :return Sample: The requested Sample object
+        """
+
+        samples = self.get_samples(sample_name)
+        if len(samples) > 1:
+            _LOGGER.warn("More than one sample was detected; returning the first")
+
+        if len(samples) == 0:
+            raise ValueError("Project has no sample named {name}.".format(name=sample_name))
 
-        Provide the project with a name, taken to be the name of the folder
-        in which its configuration file lives.
+        return samples[0]
+
+
+    def get_samples(self, sample_names):
+        """
+        Returns a list of sample objects given a list of sample names
 
-        :param str path_config_file: path to the project's configuration file.
-        :return str: name of the configuration file's folder, to name project.
+        :param list sample_names: A list of sample names to retrieve
+        :return: List: A list of Sample objects
```

Comment:
`List` should be `list`, and ideally list[Sample]. Also there's an extra colon between the tag and the type

Reply:
yeah that's funky. no idea how that got through...